Update NativeLibraryInterop docs (#443)
MicrosoftDocs/CommunityToolkit: docs/maui/native-library-interop/get-started.md

1. Update the contents of _template/macios/NewBinding.MaciOS.Binding/ApiDefinition.cs_ by replacing it with the contents of _template/macios/native/NewBinding/bin/Release/net8.0-ios/sharpie/NewBinding/ApiDefinitions.cs_.
1. Run `dotnet build` from _template/macios/NewBinding.MaciOS.Binding_ again.

#### API Definition: Android

On the native side, make updates in _template/android/native/app/src/main/java/com/example/newbinding/DotnetNewBinding.java_:

1. Add an import statement to import the native library you just added.
1. Write the API definitions for the native library APIs of interest.
1. Ensure the Android Studio project builds successfully and you are satisfied with the APIs.

Back on the .NET side, we are now ready to interop with the native library:
1. Run `dotnet build` from _template/android/NewBinding.Android.Binding_ to test everything is plugged in correctly and good to go. (Note: This step will require that you have JDK 17 installed)

> 1. Reference the Android binding dependency by adding the following code to _template/sample/MauiSample.csproj_. Replace _{yourNewBindingLibrary.aar}_ with your AndroidLibrary .aar file name. (Note: The gradle/maven dependencies often need to be explicitly referenced, as they are not automatically bundled into your library. The _build.gradle.kts_ file is configured to copy depencies into a bin/outputs/deps folder for you to reference in your .NET sample)

Reviewer: ```suggestion
1. Reference any Android binding dependencies by adding the following code to _template/sample/MauiSample.csproj_. Replace _{yourDependencyLibrary.aar}_ with any required .aar files from the library you are binding. (Note: The gradle/maven dependencies often need to be explicitly referenced, as they are not automatically bundled into your library. The _build.gradle.kts_ file is configured to copy dependencies into a bin/outputs/deps folder for you to reference in your application)
```
Small update to try to improve clarity that certain .aar files from the library being bound may need to be explicitly included in the application.

```xml
<ItemGroup Condition="$(TargetFramework.Contains('android'))">

>     <AndroidLibrary Include="..\android\native\newbinding\build\outputs\deps\{yourNewBindingLibrary.aar}">

Reviewer: ```suggestion
    <AndroidLibrary Include="..\android\native\newbinding\bin\outputs\deps\{yourNewBindingLibrary.aar}">
```
Small update for the latest path here

Author: Hmm... for me, it is showing up in the build folder? I tried again from the template directly in the latest version of the repo
<img width="348" alt="image" src="https://github.com/user-attachments/assets/16bee959-6f9f-49f4-8db7-3c129d8504f7">

Reviewer: Oh sorry I missed an update on the template, there is a ${buildDir} property that can be used over there which should update this path to something like `bin\Release\net8.0-android\outputs\deps`
